source/Cosmos.Debug.Hosts/VMware.cs

```diff
@@ -88,22 +88,44 @@ public class VMware : Host {
       }
       xPSI.UseShellExecute = false;  //must be true to allow elevate the process, sometimes needed if vmware only runs with admin rights
       mProcess.EnableRaisingEvents = true;
-      mProcess.Exited += delegate(object aSender, EventArgs e) {
-        if (OnShutDown != null) {
-          OnShutDown(aSender, e);
-        }
-      };
+      mProcess.Exited += ExitCallback;
       mProcess.Start();
     }
 
+    private void ExitCallback(object sender, EventArgs e)
+    {
+      if (null != OnShutDown)
+      {
+        try
+        {
+          OnShutDown(sender, e);
+        }
+        catch
+        {
+        }
+      }
+    }
+
     public override void Stop() {
-      //using (var xHost = new VMWareVirtualHost()) {
-      //  ConnectToVMWare(xHost);
-      //  using (var xMachine = xHost.Open(mVmxPath)) {
-      //    xMachine.PowerOff();
-      //  }
-      //  xHost.Close();
-      //}
+      if (null != mProcess)
+      {
+        try
+        {
+          //TODO: Close VMWare properly
+
+          //Force Kill VMWare
+          mProcess.Kill();
+
+          //kil vmware-vmx.exe
+          foreach (var process in Process.GetProcessesByName("vmware-vmx.exe"))
```

Comment:
This should be changed in the future as it could close other vms that the user has open

Reply:
Yes I thought about this, that's why I put a TODO

Reply:
https://github.com/CosmosOS/Cosmos/issues/2259